src/test/java/org/jenkinsci/plugins/matrixauth/ConfigurationAsCodeTest.java

```diff
@@ -0,0 +1,36 @@
+package org.jenkinsci.plugins.matrixauth;
+
+import hudson.security.ProjectMatrixAuthorizationStrategy;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import jenkins.model.Jenkins;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by mads on 2/22/18.
+ */
+public class ConfigurationAsCodeTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    public void should_configure_permissions() throws Exception {
```

Comment:
Missing a test case for exporting the config here:
https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/PLUGINS.md#model-export-test

The current released support for matrix auth plugin and export with configuration is broken.
It successfully exports but won't accept the exported version and fails with illegalstateexception

Reply:
would be nice indeed, even we don't have a yaml assertion library to test actual export. Also, export being highly experimental and missing support from many components, including jenkins-core, I won't consider this a blocker.

Reply:
not a blocker but in general when export doesn't stacktrace it does work, unfortunately in what's released currently it doesn't :(

Reply:
@ndeloof I've just tested this branch,
Is it expected that export would export it in this form?
```
  authorizationStrategy:
    globalMatrix:
      grantedPermissions:
      - "Permission[class hudson.model.View,Delete]:authenticated"
      - "Permission[class hudson.model.Computer,Connect]:authenticated"
      - "Permission[class hudson.model.Run,Delete]:authenticated"
      - "Permission[class hudson.model.Computer,Create]:authenticated"
      - "Permission[class hudson.model.View,Configure]:authenticated"
      - "Permission[class hudson.model.Computer,Build]:authenticated"
      - "Permission[interface hudson.model.Item,Configure]:authenticated"
      - "Permission[class hudson.model.Hudson,Administer]:authenticated"
      - "Permission[class hudson.model.Hudson,Administer]:300e771f-856c-45cc-b899-40d78281e9c1"
```

Reply:
No. We need to provide a custom converter for hudson.security.Permission. Initial adapter implementation has been designed for this specific plugin (before export was implemented) but this is a more generic need.

Reply:
to be done in this PR or separate? I've had a bit of a look at the code but doesn't look easy from what I can see, I assume its a change to `ConifigurationAsCode#toYaml`